### ISSUE TITLE
xt::linalg multiplication always yields 0 vectors for random input array

### ISSUE THREAD

**Issue by CermakM** [QuantStack/xtensor-blas#93]
When mulitplying two arrays, the result is always vector of `0.` (shape agrees, tho).

```c++
auto x = xt::random::randn<double>({784});
auto weights = xt::xarray<double> ({1});  // should perform identity

auto res = xt::linalg::outer(x, xt::transpose(weights));
auto res = xt::linalg::tensordot(x, xt::transpose(weights), 0);   // same result, also same without transpose
```

The output was observed in Jupyter Notebook... however, when compiled and executed in my program, I get even stranger error:

`Parameter 8 to routine cblas_dger   was incorrect`

Quick help appreciated!

**EDIT:** This seems to be the case only for `xt::random::randn`. When using `xt::arange`, the output is as expected.

**Comment by wolfv** [comment]
Thanks for the bug report. 
On a first glance, assigning `x` to a memory container *should* fix this issue for now (or using eval).

```
xarray<double> x = xt::random::randn<double>({784});
```

But we need to internally do this in tensordot.

**Comment by CermakM** [comment]
Okay, will give it a shot and hope for a quick fix. Thanks!

**Comment by wolfv** [comment]
there was indeed another problem. Fixed here https://github.com/QuantStack/xtensor-blas/pull/104

**Comment by wolfv** [comment]
we've released xtensor-blas 0.16 with a fix for this issue.
